packages/api/src/utils/stripe.js

```diff
@@ -256,7 +258,26 @@ export class StripeCustomersService {
       name: options?.name,
       email: options?.email
     })
-    await this.userCustomerService.upsertUserCustomer(user.id.toString(), createdCustomer.id)
+
+    try {
+      await this.userCustomerService.insertUserCustomer(user.id.toString(), createdCustomer.id)
+    } catch(error) {
+      // due to potential race conditions, we want to ensure we only return an actual associated stripe customer
+      // and remove stripe customers that we fail to associate
+      if (hasOwnProperty(error, "code") && error.code === "CONSTRAINT_ERROR_DB") {
```

Comment:
where does this error come from?

Reply:
this comes from postgrest.  I have the method checking for codes that start with '23' ([constraint violation codes](https://www.postgresql.org/docs/current/errcodes-appendix.html))